packages/react-core/src/components/ClipboardCopy/ClipboardCopy.tsx

```diff
@@ -111,6 +111,12 @@ export class ClipboardCopy extends React.Component<ClipboardCopyProps, Clipboard
     }
   };
 
+  componentWillUnmount = () => {
+    if (this.timer) {
+      window.clearTimeout(this.timer);
```

Comment:
Great catch! 
Does this mean that the `this.timer = null;` on line 204 and 357 is unnecessary?

Reply:
@nicolethoen that's a great question.
These lines were meant for a different case:
1) If the user clicked on the copy button we set a timer that on timeout we will close the "copied" popover
2) but what happens if the user clicks again on the copy button and the timer didn't timeout. A new timer will be created unnecessarily and we will have a flaky behavior in the popover display (it will be closed after `switchDelay` from the first click).
3) So what we did is check if there is a timer. Stop it and release it from the memory. and run a new timer.

In other words, these lines are necessary for a behavior of the popover and not covering the case where the component is unmounted and after the timeout we are changing the component state (we "cannot" change its state because it's not mounted!).

I hope I explained that well.
Thank you for the question - it was fun. 😄

Reply:
Fabulous explanation! Thank you so much!